runbot/models/build_error.py

```diff
@@ -242,6 +259,42 @@ def action_view_errors(self):
             'target': 'current',
         }
 
+    def action_infer_qualifiers(self):
+        for record in self:
+            all_qualifiers = [r.qualifiers.dict for r in record.error_content_ids]
+            common_keys = set.intersection(*map(set, [q.keys() for q in all_qualifiers]))
+            if common_keys:
+                infered_qualifiers = dict()
+                for k in common_keys:
+                    values = {q.get(k) for q in all_qualifiers}
+                    if len(values) == 1:
+                        infered_qualifiers[k] = values.pop()
+                if infered_qualifiers:
+                    record.qualifiers = infered_qualifiers
```

Comment:
I think this could be easier to read and maintain (untested)
```suggestion
            qualifiers = defaultdict(set)
            key_count = defaultdict(int)
            for content in record.error_content_ids:
                for key, value in content.qualifiers.dict.items():
                    qualifiers[key].add(value)
                    key_count[key] += 1
            record.qualifiers = {k: v[0] for k, v in qualifiers.items() if len(v) == 1 and key_count[k] == len(record.error_content_ids)}
```